httputil/client_test.go

```diff
@@ -46,3 +47,78 @@ func TestGetRequestIP(t *testing.T) {
 		})
 	}
 }
+
+func TestDefaultHttpClient(t *testing.T) {
+	client := httputil.DefaultHttpClient()
+	require.NotNil(t, client)
+	require.Equal(t, httputil.DefaultRequestTimeout, client.Timeout)
+
+	transport, ok := client.Transport.(*http.Transport)
+	require.True(t, ok)
+	require.True(t, transport.DisableKeepAlives)
+	require.Equal(t, httputil.DefaultMaxConnsPerHost, transport.MaxConnsPerHost)
+	require.Equal(t, httputil.DefaultMaxIdleConnsPerHost, transport.MaxIdleConnsPerHost)
+	require.Equal(t, httputil.DefaultIdleConnTimeout, transport.IdleConnTimeout)
+	require.Equal(t, httputil.DefaultTransport(), transport)
+}
+
+func TestNewHttpClientWithOptions(t *testing.T) {
+	customTransport := &http.Transport{
+		DisableKeepAlives: false,
+		Proxy:             http.ProxyURL(nil),
+	}
+	testCases := []struct {
+		name          string
+		options       []httputil.HttpClientOptions
+		expectedCheck func(*testing.T, *http.Client)
+	}{
+		{
+			name:    "No options",
+			options: nil,
+			expectedCheck: func(t *testing.T, client *http.Client) {
+				require.Equal(t, httputil.DefaultRequestTimeout, client.Timeout)
+				transport, ok := client.Transport.(*http.Transport)
+				require.True(t, ok)
+				require.True(t, transport.DisableKeepAlives)
+			},
+		},
+		{
+			name:    "WithTimeout",
+			options: []httputil.HttpClientOptions{httputil.WithTimeout(5 * time.Second)},
+			expectedCheck: func(t *testing.T, client *http.Client) {
+				require.Equal(t, 5*time.Second, client.Timeout)
+				transport, ok := client.Transport.(*http.Transport)
+				require.True(t, ok)
+				require.True(t, transport.DisableKeepAlives)
+			},
+		},
+		{
+			name:    "WithTransport",
+			options: []httputil.HttpClientOptions{httputil.WithTransport(customTransport)},
+			expectedCheck: func(t *testing.T, client *http.Client) {
+				require.NotNil(t, client)
+				require.Equal(t, httputil.DefaultRequestTimeout, client.Timeout)
+				require.Same(t, customTransport, client.Transport)
+			},
+		},
+		{
+			name: "Multiple options",
+			options: []httputil.HttpClientOptions{
+				httputil.WithTimeout(15 * time.Second),
+				httputil.WithTransport(customTransport),
+			},
+			expectedCheck: func(t *testing.T, client *http.Client) {
+				require.NotNil(t, client)
+				require.Equal(t, 15*time.Second, client.Timeout)
+				require.Same(t, customTransport, client.Transport)
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			client := httputil.NewHttpClient(tc.options...)
+			tc.expectedCheck(t, client)
+		})
+	}
+}
```

Comment:
We can add an example like:

```go
func Example() {
	transport := httputil.DefaultTransport() // no need to use .Clone() since a new transport is built each time
	transport.ForceAttemptHTTP2 = true
	client := httputil.NewHttpClient(httputil.WithTransport(transport))
	req, err := http.NewRequest("GET", "https://example.com", nil)
	if err != nil {
		panic(err)
	}
	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	_ = resp.Body.Close()
}
```